src/main/java/net/juniper/netconf/Device.java

```diff
@@ -838,14 +839,48 @@ public XML getRunningConfig(String configTree) throws SAXException,
         return this.netconfSession.getRunningConfig(configTree);
     }
 
-    public XML getRunningConfigAndState(String filter) throws IOException, SAXException {
+    /**
+     * Retrieve the running configuration, or part of the configuration.
+     *
+     * @param xpathFilter example <code><filter xmlns:model='urn:path:for:my:model' select='/model:*'></filter></code>
+     * @return configuration data as XML object.
+     * @throws java.io.IOException      If there are errors communicating with the netconf server.
+     * @throws org.xml.sax.SAXException If there are errors parsing the XML reply.
+     */
+    public XML getRunningConfigAndState(String xpathFilter) throws IOException, SAXException {
+        if (netconfSession == null) {
+            throw new IllegalStateException("Cannot execute RPC, you need to " +
+                    "establish a connection first.");
+        }
+        return this.netconfSession.getRunningConfigAndState(xpathFilter);
+    }
+
+
+    /**
+     * Run the <get-data> call to netconf server and retrieve data as an XML.
+     *
+     * @param xpathFilter example <code><filter xmlns:model='urn:path:for:my:model' select='/model:*'></filter></code>
+     * @param datastore running, startup, candidate, or operational
+     * @return configuration data as XML object.
+     * @throws java.io.IOException      If there are errors communicating with the netconf server.
+     * @throws org.xml.sax.SAXException If there are errors parsing the XML reply.
+     */
+    public XML getData(String xpathFilter, String datastore) throws IOException, SAXException {
         if (netconfSession == null) {
             throw new IllegalStateException("Cannot execute RPC, you need to " +
                     "establish a connection first.");
         }
-        return this.netconfSession.getRunningConfigAndState(filter);
+        if (datastore != null) switch (datastore.trim().toLowerCase(Locale.ROOT)) {
```

Comment:
I can't help but think datastore should be an enum rather than a free-format string that only has four valid options.

Reply:
@GregDThomas - I agree in general, but for small numbers of options (which are strings anyway), I wasn't sure if an enum was necessary. 

Instead, a localized string (that's why I threw in `Locale.ROOT)`. I figured for this situation, its more of a stylistic thing. There are a [small finite number](https://tools.ietf.org/id/draft-ietf-netmod-revised-datastores-08.html#datastore-model) of these states, I wasn't sure if programming in an enum was wanted.

That said, I can easily put together an enum here. I have no problem doing that. As I think about it, I probably souldn't have used `Locale.ROOT` and used `Locale.US`

Reply:
a) I'm not an approved committer, just a user of the library, so feel free to ignore!

b) "There are a small finite number of these states" - that's exactly the sort of use case for an Enum, IMHO. Also makes it quite clear to the caller of the API what values would be valid. 

If it were me, I'd use something like ...

```
@Getter
@RequiredArgsConstructor
public enum Datastore

    RUNNING("running"),
    CANDIDATE("candidate"),
    STARTUP("startup"),
    INTENDED("intended");

    public final String tag;   
```
Then you could use `datastore.getTag()`

Reply:
@GregDThomas - Yeah I agree. Its stylistic and also, I'm glad you are using Lombok here for this. Much less code. I will implement an enum now.

Reply:
Okay, Datastore enum is made. I put it in the `element` package as it seemed aligned with the other classes in there, in that its based off RFC defined objects (such as RpcReply, Hello, etc.